src/AutoclassesBundle/Command/DumpTableConfCommand.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace ChameleonSystem\AutoclassesBundle\Command;
+
+use ChameleonSystem\AutoclassesBundle\TableConfExport\TableConfExporterInterface;
+use ChameleonSystem\DataAccessBundle\ChameleonSystemDataAccessBundle;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\HttpKernel\Config\FileLocator;
+
+class DumpTableConfCommand extends Command
+{
+    public function __construct(
+        private readonly TableConfExporterInterface $tableConfExporter,
+        private readonly FileLocator $fileLocator,
+        private readonly array $tableClassMapping = [],
+    ) {
+        parent::__construct();
+    }
+    protected function configure()
+    {
+        $this->setName('chameleon_system:autoclasses:dump')
+            ->setDescription('Export the table configurations as doctrine data models');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $tables = $this->tableConfExporter->getTables();
+
+
+        $tableNamespaceMapping = $this->getTableNamespaceMapping();
+        foreach ($tables as $table) {
+            $tableConfig = $this->getTableConfig($table->name);
+            if (null === $tableConfig) {
+                $output->writeln(sprintf('<warning>No Config for %s found - using default</warning>', $table->name));
+                $tableConfig = [
+                    'targetDir' => '@AppBundle/src/Entity',
+                    'configDir' => 'AppBundle/config/doctrine',
+                    'namespace' => '\\AppBundle\\Entity',
+                    'tables' => [],
+                ];
+            }
+            $output->write(sprintf('<info>Exporting %s...</info>',$table->name));
+
+            try {
+
+                if ('@' === substr( $tableConfig['targetDir'], 0, 1)) {
+                    $bundle = substr($tableConfig['targetDir'], 0, strpos($tableConfig['targetDir'], '/'));
+                    $bundleBasePath = $this->fileLocator->locate($bundle);
+
+                    $tableConfig['targetDir'] = str_replace(sprintf('%s/', $bundle),$bundleBasePath, $tableConfig['targetDir']);
+                    $tableConfig['configDir'] = str_replace(sprintf('%s/', $bundle),$bundleBasePath, $tableConfig['configDir']);
+                }
+
+                $targetDir = $tableConfig['targetDir'];
+                $mappingDir = $tableConfig['configDir'];
+                if (false===is_dir($targetDir)) {
+                    if (!mkdir($targetDir, 0755, true) && !is_dir($targetDir)) {
```

Comment:
More than one request may arrive at that line at the same time. Since mkdir returns false if the directory already exists, and since the first is_dir is not atomically combined with mkdir, we check after a failed mkdir if perhapse another request created the directory. If so, we are good. If not, then - and only then - do we need to fail ('false' here has to different meanings - and that is our way of dealing with it)